src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -82,6 +99,13 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     host = new HostString(host.Host);
                 }
 
+                if (!Permanent.HasValue)
+                {
+                    //if MvcOption.requireHttpsPermanent is not null use it, otherwise use default(bool)
+                    Permanent = optionsAccessor.Value.RequireHttpsPermanent.HasValue ? optionsAccessor.Value.RequireHttpsPermanent.Value : default(bool);
```

Comment:
Oh... yes the short form is more clean..

Ok good i do RequireHttpsPermanent not null so i can do this 
Permanent = Permanent ?? optionsAccessor.Value.RequireHttpsPermanent;
instead of the intire if statement